src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -532,16 +526,15 @@ private static int GetFloatingPointMaxDigitsAndPrecision(char fmt, ref int preci
             }
 
             char fmt = ParseFormatSpecifier(format, out int precision);
-            byte* pDigits = stackalloc byte[DoubleNumberBufferLength];
-
             if (fmt == '\0')
             {
                 // For back-compat we currently specially treat the precision for custom
                 // format specifiers. The constant has more details as to why.
                 precision = DoublePrecisionCustomFormat;
             }
 
-            NumberBuffer number = new NumberBuffer(NumberBufferKind.FloatingPoint, pDigits, DoubleNumberBufferLength);
+            byte* pDigits = stackalloc byte[DoubleNumberBufferLength];
+            NumberBuffer number = new NumberBuffer(NumberBufferKind.FloatingPoint, new Span<byte>(pDigits, DoubleNumberBufferLength));
```

Comment:
```suggestion
            NumberBuffer number = new NumberBuffer(NumberBufferKind.FloatingPoint, stackalloc byte[DoubleNumberBufferLength]);
```
Is there a reason why this has the pDigits temp local? The local was removed in other places.

Reply:
This and two other places couldn't remove it due to C# ref lifetime rules (the builder is passed in as an argument and then both the builder and this are passed by ref to another method).